Modules/System/SharePoint/src/SharePointClientImpl.Codeunit.al

```diff
@@ -0,0 +1,581 @@
+codeunit 9101 "SharePoint Client Impl."
+{
+    Access = Internal;
+
+    var
+        SharePointUriBuilder: Codeunit "SharePoint Uri Builder";
+        SharePointRequestHelper: Codeunit "SharePoint Request Helper";
+        SharePointOperationResponse: Codeunit "SharePoint Operation Response";
+        Authorization: Interface "SharePoint Authorization";
+        ReadResponseFailedErr: Label 'Could not read response.';
+        IncorrectResponseErr: Label 'Incorrect response.';
+
+    procedure Initialize(BaseUrl: Text; Auth: Interface "SharePoint Authorization")
+    begin
+        SharePointUriBuilder.Initialize(BaseUrl, 'Web');
+        SharePointUriBuilder.ResetPath();
+        Authorization := Auth;
+    end;
+
+    procedure Initialize(BaseUrl: Text; Namespace: Text; Auth: Interface "SharePoint Authorization")
+    begin
+        SharePointUriBuilder.Initialize(BaseUrl, Namespace);
+        SharePointUriBuilder.ResetPath();
+        Authorization := Auth;
+    end;
+
+    procedure GetDiagnostics(): Codeunit "SharePoint Diagnostics"
+    begin
+        exit(SharePointOperationResponse.GetDiagnostics());
+    end;
+
+    local procedure GetRequestDigest(BaseUrl: Text): Text
+    var
+        NewSharePointOperationResponse: Codeunit "SharePoint Operation Response";
+        NewSharePointUriBuilder: Codeunit "SharePoint Uri Builder";
+        Context: JsonToken;
+        Result: Text;
+    begin
+        NewSharePointUriBuilder.Initialize(BaseUrl, 'contextinfo');
+        SharePointRequestHelper.SetAuthorization(Authorization);
+        NewSharePointOperationResponse := SharePointRequestHelper.Post(NewSharePointUriBuilder);
+        if NewSharePointOperationResponse.GetResultAsText(Result) then begin
+
+            Context.ReadFrom(Result);
+
+            if not Context.AsObject().Get('d', Context) then
+                Error(IncorrectResponseErr);
```

Comment:
This is using the same error message for each error, maybe do a logmessage to telemetry to give more context if this is something that be resolved by the developer etc?

Reply:
ok, done